internal/models/linking.go

```diff
@@ -78,8 +78,12 @@ func DetermineAccountLinking(tx *storage.Connection, provider, sub string, email
 			return AccountLinkingResult{}, terr
 		}
 
-		if terr := tx.Q().Eager().Where("email in (?) and is_sso_user is false", emails).All(&similarUsers); terr != nil {
-			return AccountLinkingResult{}, terr
+		if !strings.HasPrefix(provider, "sso:") {
```

Comment:
Hmm if there's an `sso:` prefix, should we search for `is_sso_user is true` instead? I think it's still necessary to see if there's already an existing sso user in the `auth.users` table right? 

Context: we had to check the user's table as well because in some cases, a user can exist without any identities (admin create / update user, invite user) because that was overlooked. 

But maybe this is not necessary for sso users because we can be sure that all sso users will have a corresponding sso identity right?

Reply:
> Hmm if there's an sso: prefix, should we search for is_sso_user is true instead? I think it's still necessary to see if there's already an existing sso user in the auth.users table right?

Originally that's what I thought as well. But the problem is that SSO users can have duplicate emails, so we would not be able to decide what account to link to. That's why I changed it to only look for similar users if the _identity being linked_ has nothing to do with SSO.

And finally, you're right that there would always be identities for SSO users, so the linking algorithm doesn't need the users table to decide.